test/jb_projects/claim_handle.test.js

```diff
@@ -0,0 +1,290 @@
+      );
+
+    await jbProjectsStore
+      .connect(projectOwner)
+      .transferHandleOf(
+        PROJECT_ID_1,
+        deployer.address,
+        ethers.utils.formatBytes32String(PROJECT_HANDLE_3),
+      );
+
+    await expect(
+      jbProjectsStore
+        .connect(projectOwner)
+        .claimHandle(
+          ethers.utils.formatBytes32String(PROJECT_HANDLE_1),
+          deployer.address,
+          PROJECT_ID_2,
+        ),
+    ).to.be.revertedWith('Operatable: UNAUTHORIZED');
+  });
+
+  it(`Can't claim if it is after expiration date`, async function () {
+    const { projectOwner, deployer, jbProjectsStore } = await setup();
+
+    await jbProjectsStore
+      .connect(deployer)
+      .createFor(
+        projectOwner.address,
+        ethers.utils.formatBytes32String(PROJECT_HANDLE_1),
+        METADATA_CID,
+      );
+
+    await jbProjectsStore
+      .connect(deployer)
+      .createFor(
+        deployer.address,
+        ethers.utils.formatBytes32String(PROJECT_HANDLE_2),
+        METADATA_CID,
+      );
+
+    await jbProjectsStore
+      .connect(projectOwner)
+      .transferHandleOf(
+        PROJECT_ID_1,
+        deployer.address,
+        ethers.utils.formatBytes32String(PROJECT_HANDLE_3),
+      );
+
+    let tx = await jbProjectsStore
+      .connect(projectOwner)
+      .challengeHandle(ethers.utils.formatBytes32String(PROJECT_HANDLE_2));
+
+    await fastForward(tx.blockNumber, ethers.BigNumber.from(12536000));
+
+    await expect(
+      jbProjectsStore
+        .connect(deployer)
+        .claimHandle(
+          ethers.utils.formatBytes32String(PROJECT_HANDLE_2),
+          deployer.address,
+          PROJECT_ID_2,
+        ),
+    ).to.be.revertedWith('0x0c: UNAUTHORIZED');
+  });
+
+  it(`Can't claim handle and assign to inexistent or not owned projectId`, async function () {
+    const { projectOwner, deployer, addrs, jbProjectsStore, mockJbOperatorStore } = await setup();
+
+    await jbProjectsStore
+      .connect(deployer)
+      .createFor(
+        projectOwner.address,
+        ethers.utils.formatBytes32String(PROJECT_HANDLE_1),
+        METADATA_CID,
+      );
+
+    let handleReceiver = addrs[0];
+    await jbProjectsStore
+      .connect(projectOwner)
+      .transferHandleOf(
+        PROJECT_ID_1,
+        handleReceiver.address,
+        ethers.utils.formatBytes32String(PROJECT_HANDLE_2),
+      );
+
+    await mockJbOperatorStore.mock.hasPermission
+      .withArgs(
+        handleReceiver.address,
+        deployer.address,
+        PROJECT_ID_1,
+        CLAIM_HANDLE_PERMISSION_INDEX,
+      )
+      .returns(true);
+
+    await expect(
+      jbProjectsStore
+        .connect(handleReceiver)
+        .claimHandle(
+          ethers.utils.formatBytes32String(PROJECT_HANDLE_1),
+          projectOwner.address,
+          PROJECT_ID_2,
+        ),
+    ).to.be.reverted;
+  });
+
+  it(`Can't claim handle even if its recipient who does it but without permissions`, async function () {
```

Comment:
not sure about this test.

The two `requirePermission`'s block for `claimHandle` make sure that the address claiming is both the address that the handle was transferred to, and the owner of the project that the claimed handle is being set for.

Reply:
Correct, but what if the handle is transferred and later the permission is removed? we need to test that when the permission is revoked, the recipient of the handle cannot claim it.

Reply:
I still think we're slightly off the same page.

`requirePermission(_transferAddress, _projectId, JBOperations.CLAIM_HANDLE)` this line means "the msg.sender must either be the _transferAddress, or it must have been given the JBOperations.CLAIM_HANDLE permission from the _transferAddress".

we need to test that when the permission is given to an operator by the _transferRecipient, the operator can claim it the handle if it is also the owner of project having the handle claimed onto or an operator thereof. The _transferRecipient can also claim at any time if it is also the project owner of the handle being claimed onto or is an operator.

Reply:
So what is missing is the operator's permissions test, right? the other two tests are already covered with previous tests.

Reply:
It happens I was initializing JBProjects without the mock contract for permissions mocking, that was driving me crazy. Now it should cover the use cases of:

1. Test handleReceiver claim --> success
2. Test operator claim (not owner of handle or receiver) --> fail
3. Test operator with permissions (not owner of handle or receiver) --> success